src/hooks/useTxHistory.ts

```diff
@@ -11,24 +13,30 @@ const useTxHistory = (
   error?: string
   loading: boolean
 } => {
+  const router = useRouter()
+  const hasFilter = hasTxFilterQuery(router.query)
+
   const { safe, safeAddress, safeLoaded } = useSafeInfo()
   const { chainId } = safe
 
   // If pageUrl is passed, load a new history page from the API
   const [page, error, loading] = useAsync<TransactionListPage>(
     () => {
-      if (!pageUrl || !safeLoaded) return
-      return getTransactionHistory(chainId, safeAddress, pageUrl)
+      if (!safeLoaded || !(hasFilter || pageUrl)) return
+
+      return hasFilter
+        ? getFilteredTxHistory(chainId, safeAddress, router.query, pageUrl)
```

Comment:
`getFilteredTxHistory` could accept already the actual filter object and not the entire query.

Reply:
This ties into the above comment. We need the `TxFilterFormType` from the `query`. It would require a lot more excess code that includes a few type guards. I can change it but don't think it's worth it. Wdyt?